libzkbob-rs/src/client/mod.rs

```diff
@@ -394,19 +405,14 @@ where
                 let out_notes = outputs
                     .iter()
                     .map(|dest| {
-                        let (to_d, to_p_d, pool) = parse_address::<P>(&dest.to, &self.params)?;
-
-                        if pool.is_none() || pool.unwrap() == self.pool {
-                            output_value += dest.amount.to_num();
-                            Ok(Note {
-                                d: to_d,
-                                p_d: to_p_d,
-                                b: dest.amount,
-                                t: rng.gen(),
-                            })
-                        } else {
```

Comment:
Is it safe to remove this check?

Reply:
Yes it's safe. Currently `libzkbob-rs` know nothing about pools with other IDs. In case the address provided by the user belongs to the another pool the validation (in `parse_address`) will fail with an `InvalidChecksum` error and `unwrap` will also fail. Checking which pool zk address belongs to is the client library responsibility field now